src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2739,6 +2739,8 @@ protected function castAttribute($key, $value)
 			case 'array':
 			case 'json':
 				return json_decode($value, true);
+			case 'collection':
+				return new Collection(json_decode($value, true));
```

Comment:
Use `$this->newCollection`, so that it'll use the user-provided collection class.